internal/service/streamprocessor/resource_test.go

```diff
@@ -0,0 +1,350 @@
+					checkExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "instance_name"),
+					resource.TestCheckResourceAttrSet(resourceName, "processor_name"),
+					resource.TestCheckResourceAttr(resourceName, "processor_name", processorName),
+					resource.TestCheckResourceAttr(resourceName, "state", "STARTED"),
+				),
+			},
+			{
+				Config: streamProcessorConfigWithSampleConnection(projectID, instanceName, processorName, "STOPPED"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "instance_name"),
+					resource.TestCheckResourceAttrSet(resourceName, "processor_name"),
+					resource.TestCheckResourceAttr(resourceName, "processor_name", processorName),
+					resource.TestCheckResourceAttr(resourceName, "state", "STOPPED"),
+				),
+			},
+		}})
+}
+
+func TestAccStreamProcessorRS_clusterType(t *testing.T) {
+	var (
+		projectID, clusterName = acc.ClusterNameExecution(t)
+		processorName          = "new-processor"
+		instanceName           = acc.RandomName()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroyStreamProcessor,
+		Steps: []resource.TestStep{
+			{
+				Config: streamProcessorConfigWithClusterConnection(projectID, clusterName, instanceName, processorName, "STARTED"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "instance_name"),
+					resource.TestCheckResourceAttrSet(resourceName, "processor_name"),
+					resource.TestCheckResourceAttr(resourceName, "processor_name", processorName),
+					resource.TestCheckResourceAttr(resourceName, "state", "STARTED"),
+					resource.TestCheckResourceAttrSet(resourceName, "stats"),
+				),
+			},
+		}})
+}
+
+func TestAccStreamProcessorRS_failWithInvalidStateOnCreation(t *testing.T) {
+	var (
+		projectID     = acc.ProjectIDExecution(t)
+		processorName = "new-processor"
+		instanceName  = acc.RandomName()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroyStreamProcessor,
+		Steps: []resource.TestStep{
+			{
+				Config:      streamProcessorConfigWithSampleConnection(projectID, instanceName, processorName, "STOPPED"),
+				ExpectError: regexp.MustCompile("When creating a stream processor, the only valid states are CREATED and STARTED"),
+			},
+		}})
+}
+
+func streamProcessorConfigWithSampleConnection(projectID, instanceName, processorName, state string) string {
```

Comment:
nit: In recent refactors I see we have gone with approach of removing verbosity in the name of these config and functions, for examples here would be `configWIthSampleConnection` (for reference you can see resources like alert_configuration, backup_compliance_policy, advanced_cluster). We could adjust the scaffolding template as well to avoid confusion moving forward.

Reply:
Done!